core/capabilities/ccip/ccipevm/extraargscodec.go

```diff
@@ -1,16 +1,60 @@
 package ccipevm
 
 import (
+	"fmt"
+
+	chainsel "github.com/smartcontractkit/chain-selectors"
+
 	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
 )
 
+const (
+	EVMExtraArgsKey     = "gasLimit"
+	SVMComputeUnitKey   = "ComputeUnits"
+	SVMAccountBitmapKey = "AccountIsWritableBitmap"
+	SVMAllowOOEKey      = "AllowOutOfOrderExecution"
+	SVMTokenReceiverKey = "TokenReceiver"
+	SVMAccountListKey   = "Accounts"
+)
+
 type ExtraArgsCodec struct{}
 
 func NewExtraArgsCodec() ExtraArgsCodec {
 	return ExtraArgsCodec{}
 }
 
 func (ExtraArgsCodec) DecodeExtraData(extraArgs cciptypes.Bytes, sourceChainSelector cciptypes.ChainSelector) (map[string]any, error) {
-	// Not implemented and not return error
-	return nil, nil
+	family, err := chainsel.GetSelectorFamily(uint64(sourceChainSelector))
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode extra data, %w", err)
+	}
+
+	switch family {
+	case chainsel.FamilyEVM:
+		gas, err1 := decodeExtraArgsV1V2(extraArgs)
+		if err1 != nil {
+			return nil, fmt.Errorf("failed to decode EVM extra data, %w", err)
+		}
+
+		return map[string]any{
+			EVMExtraArgsKey: gas,
+		}, nil
```

Comment:
I'd be tempted to do some minor refactoring here so you can write it like this:
```suggestion
	case chainsel.FamilyEVM:
		return decodeExtraArgsV1V2(extraArgs)
```

Maybe even some moderate refactoring to avoid having SVM code in the EVM package.

Reply:
`decodeExtraArgsV1V2` returns `*big.int`. The return type needs to be `map[string]any`

Reply:
The way I understand, the SVMExtraArgsv1 is defined for EVM onchain, and requires abi decoding, which is all EVM

Reply:
Yes, I said there would be minor refactoring.